src/MetadataManager/pathToHash.ts

```diff
@@ -0,0 +1,273 @@
+      if (fs.lstatSync(path).isDirectory()) {
+        for (let key in subPathToHash[1]) {
+          queue.push([key, subPathToHash[1][key], subPathToHash[2] + '/' + key]);
+        }
+      } else {
+        flatPathTohash[subPathToHash[2]] = subPathToHash[1];
+      }
+    }
+    return flatPathTohash;
+  }
+
+  public async createMetadata(flattenPathToHash: any) {
+    for (let path in flattenPathToHash) {
+      const hash = flattenPathToHash[path];
+
+      let metadata: any = await Metadata.get(hash);
+
+      if (Object.keys(metadata).length === 0) {
+        await Metadata.set(hash, {});
+
+      } else if (metadata[path]) {
+        return;
+      }
+
+      const filename: any = path.split('/').pop();
+      if (vscode.workspace.workspaceFolders === undefined) {
+        return;
+      }
+      const stats: any = await Metadata.getStats(
+          vscode.Uri.joinPath(vscode.workspace.workspaceFolders[0].uri, path));
+      metadata[path] = {};
+      metadata[path]['name'] = filename;
+      metadata[path]['file-extension'] = filename.split('.')[1];
+      metadata[path]['create-time'] = stats.birthtime;
+      metadata[path]['modified-time'] = stats.mtime;
+      metadata[path]['is-deleted'] = false;
+      await Metadata.set(hash, metadata);
+    }
+  }
+
+  public async deleteMetadata(flattenpathToHash: any) {
+    if (vscode.workspace.workspaceFolders === undefined) {
+      return;
+    }
+    const baseUri =
+        vscode.Uri.joinPath(vscode.workspace.workspaceFolders[0].uri, '.meta/hash_objects');
+    const arrayList = await vscode.workspace.fs.readDirectory(baseUri);
+    for (const array of arrayList) {
+      const hashFolderUri = vscode.Uri.joinPath(baseUri, array[0]);
+      const hashList = await vscode.workspace.fs.readDirectory(hashFolderUri);
+      for (const hashFile of hashList) {
+        const hashUri = vscode.Uri.joinPath(hashFolderUri, hashFile[0]);
+        let metadata =
+            JSON.parse(Buffer.from(await vscode.workspace.fs.readFile(hashUri)).toString());
+        const hash = array[0] + hashFile[0].split('.')[0];
+        for (const key in metadata) {
+          if (!metadata[key]['is-deleted'] && flattenpathToHash[key] !== hash) {
+            metadata[key]['is-deleted'] = true;
+          }
+        }
+        Metadata.set(hash, metadata);
+      }
+    }
+  }
+
+
+  public get(uri: vscode.Uri) {
+    const path = vscode.workspace.asRelativePath(uri);
+    const splitPath = path.split('/');
+    let pathToHash = this._map;
+
+    splitPath.forEach((data) => {
+      if (pathToHash === undefined) {
+        return undefined;
+      }
+      pathToHash = pathToHash[data];
+    });
+
+    return pathToHash;
+  }
+
+
+  public static async generateHash(uri: vscode.Uri) {
+    // TODO: Error handling
+    return crypto.createHash('sha256')
+        .update(Buffer.from(await vscode.workspace.fs.readFile(uri)).toString())
+        .digest('hex');
+  }
```

Comment:
Maybe `Hash` can be extracted as a class and related functions can be moved there